### ISSUE TITLE
TTS(tacotron2-v3) frequency min and max for man voice

### ISSUE THREAD

**Issue by EsmaeilFarhang** [espnet/espnet#1657]
Hi,
I have a dataset with man voice, I'm trained tacotron2-v3, config but the loss cureve decrease very slowly. this dataset is 19 hours wave file. frequency min and frequency max for man voice should be change?  With 120 epoch training, The sentences produced are meaningless. And they don't match with the text.
one sample is below for this sentense ('oui, j'Fteignais le gaz de l'escalier, il a poussF le verrou de sa porte. il n'y a personne avec lui ?').
![all_loss_old](https://user-images.githubusercontent.com/24624344/76139793-0df1bb80-6069-11ea-89b7-71801abb3fa1.png)

Do you have any other suggestions for better model accuracy?
![all_loss_old](https://user-images.githubusercontent.com/24624344/76139794-1649f680-6069-11ea-9b0b-93f192392a75.png)
![l1_loss_old](https://user-images.githubusercontent.com/24624344/76139796-1a761400-6069-11ea-9af6-1ef95599467c.png)
![bce_loss_old](https://user-images.githubusercontent.com/24624344/76139797-1e099b00-6069-11ea-9edb-b240378f1209.png)
![loss_old](https://user-images.githubusercontent.com/24624344/76139800-21048b80-6069-11ea-9ee2-eed93cfab823.png)
![mse_loss_old](https://user-images.githubusercontent.com/24624344/76139802-2366e580-6069-11ea-90a4-256cf844493d.png)

[test.zip](https://github.com/espnet/espnet/files/4301551/test.zip)

**Comment by kan-bayashi** [comment]
Did you check attention weight? Loss curve is not so meaningful in the case of TTS. You should the attention weight is diagonal. If the attention weight is not clear and diagonal, please check the following points:

- Trim the silence at the beginning and the end of utterance carefully. Power-based trimming scripts are available in espnet (`utils/trim_silence.sh`), but the best is to use the force alignment label (not supported in espnet).
- Check the text is correctly normalized and cleaned (number, abbreviation, case, etc). Maybe an uncased sentence is better.
- Try with reduction factor > 1 and check it works. Large reduction factor helps diagonal attention learning.

Related issues:
- #534
- #1245 
- #1360

**Comment by EsmaeilFarhang** [comment]
Hi, thank you
> Did you check attention weight?
yes, attention is poor

>     * Trim the silence at the beginning and the end of utterance carefully. Power-based trimming scripts are available in espnet (`utils/trim_silence.sh`), but the best is to use the force alignment label (not supported in espnet). I checked test.mp3, it seems to contain much silence.
i'm also for this model used trim_silence 

>     * Check the text is correctly normalized and cleaned (number, abbreviation, case, etc). Maybe an uncased sentence is better

I'm not Sure texts are clean so I'm trying to check then and train again,

>     * Try with reduction factor > 1 and check it works.  large reduction factor helps diagonal attention learning.
What should I put in this reduction factor?  and how to fine-tune that?

**Comment by EsmaeilFarhang** [comment]
I'm trying with tacotron2_v4  and Trim the silence.
 attention weight after 103 epoch is in below:
![LJ8152](https://user-images.githubusercontent.com/24624344/76834900-333db100-6843-11ea-92dc-bf16e6f9472f.gif)
![LJ8150](https://user-images.githubusercontent.com/24624344/76834882-2ae57600-6843-11ea-9dd2-de0dd37727fc.gif)
![LJ8151](https://user-images.githubusercontent.com/24624344/76834896-30db5700-6843-11ea-8866-bb94ef08605b.gif)
![LJ8153](https://user-images.githubusercontent.com/24624344/76834902-35077480-6843-11ea-8115-0ed8ffb1e38b.gif)

**Comment by kan-bayashi** [comment]
> What should I put in this reduction factor? and how to fine-tune that?

You can try 3 or 5 at first. If you cannot train well with 3 or 5, there is a problem in preprocessing, I think.

From your figure, the beginning of the audio contains something.
Could you check the figures in `exp/trim_silence/figs`?

**Comment by EsmaeilFarhang** [comment]
> > What should I put in this reduction factor? and how to fine-tune that?
> 
> You can try 3 or 5 at first. If you cannot train well with 3 or 5, there is a problem in preprocessing, I think.
> 
> From your figure, the beginning of the audio contains something.
> Could you check the figures in `exp/trim_silence/figs`?

thank you,
there are some figures:
![Screenshot from 2020-03-17 14-38-40](https://user-images.githubusercontent.com/24624344/76851226-2b3f3a80-685e-11ea-85b4-8115ad1b3196.png)
![Screenshot from 2020-03-17 14-39-38](https://user-images.githubusercontent.com/24624344/76851233-2d08fe00-685e-11ea-8be5-ae7c0623b98d.png)
![Screenshot from 2020-03-17 14-41-05](https://user-images.githubusercontent.com/24624344/76851235-2da19480-685e-11ea-82d2-0b8db11e8410.png)

**Comment by kan-bayashi** [comment]
If the number of figures like the third one is small, it is ok but if most of them are like the third one, it is better to tune the silence threshold.

**Comment by EsmaeilFarhang** [comment]
> If the number of figures like the third one is small, it is ok but if most of them are like the third one, it is better to tune the silence threshold.

Thank you,
I'm trying with v4, trimming the silence and change the reduction factor to 3.

**Comment by EsmaeilFarhang** [comment]
@kan-bayashi 
Hi 
I'm trying with tacotron.v4 config file and changed some parameters:
silence threshold= 25 DB (fine-tuned that)
normalized text 
reduction factor = 3
the attention weights are in below:
![LJ8150](https://user-images.githubusercontent.com/24624344/77255755-748cf100-6c87-11ea-8bd6-79aee8b8f76b.gif)
![LJ8151](https://user-images.githubusercontent.com/24624344/77255769-8078b300-6c87-11ea-8b50-36f211b79c99.gif)
![LJ8152](https://user-images.githubusercontent.com/24624344/77255771-8373a380-6c87-11ea-80b2-a3a733f125d7.gif)
![LJ8153](https://user-images.githubusercontent.com/24624344/77255773-88d0ee00-6c87-11ea-8f5e-2d191db8aaa8.gif)
![LJ8154](https://user-images.githubusercontent.com/24624344/77255777-94241980-6c87-11ea-8bfe-8af9ba4b06f4.gif)





after 110 epoch generated voice is so bad and words are not clear.

**Comment by kan-bayashi** [comment]
Hmm. Even if you cannot get clear attention with upsampling factor=3, it seems there is a problem in training data. How did you normalize the text? Could you show the `text` for training data?

**Comment by EsmaeilFarhang** [comment]
@kan-bayashi 
thanks.
I've used normalized text form the main dataset.
this is all text files.
[metadata.zip](https://github.com/espnet/espnet/files/4430996/metadata.zip)

**Comment by kan-bayashi** [comment]
Not directly related to your problem, #1761 fixed attention plot bug.
Please check it.
About the text, did you try with lower-case or upper-case inputs?

**Comment by EsmaeilFarhang** [comment]
> About the text, did you try with lower-case or upper-case inputs?

I'm converting all characters to lowercase and map some characters to uppercase as below:
'''
ce_dict = {'à':'A', 'â':'B', 'æ':'D', 'ç':'C', 'è':'E', 'é':'F', 'ê':'G','ë':'H', 'î':'I', 'ï':'J', 'ô':'O', 'ù':'U', 'û':'V', 'œ':'W'}
'''
this is France language and with using this mapping, I successfully trained transformer_v1 with another dataset.

**Comment by kan-bayashi** [comment]
I see.
Why don’t you use large —guided-attn-loss-lambda? Say, 10 or 100.
